src/journal/bsod.c

```diff
@@ -105,7 +107,21 @@ static int acquire_first_emergency_log_message(char **ret) {
         if (!message)
                 return log_oom();
 
-        *ret = TAKE_PTR(message);
+        r = sd_journal_get_data(j, "MESSAGE_ID", &d, &l);
+        if (r == -ENOENT) {
+                *ret_message = TAKE_PTR(message);
+                *ret_message_id = NULL;
+                return 0;
+        }
+        if (r < 0)
+                return log_error_errno(r, "Failed to read message ID from journal: %m");
+
+        message_id = memdup_suffix0((const char*)d + STRLEN("MESSAGE_ID="), l - STRLEN("MESSAGE_ID="));
```

Comment:
The cast to `const char*` is not necessary.

Also, for safety, I'd like to check if the data is prefixed with `MESSAGE_ID=`:
```c
d = memory_startswith(d, "MESSAGE_ID=", l);
if (!d)
        return log_error_errno(SYNTHETIC_ERRNO(EINVAL), "Wrong message ID.");
assert(l >= STRLEN("MESSAGE_ID="));
message_id = memdup_suffix0(d, l - STRLEN("MESSAGE_ID=");
```

Reply:
Done sir.